### ISSUE TITLE
FIX PICKING RANDOM JOKE

### ISSUE THREAD

**Issue by pitzzahh** [pitzzahh/jokes-API#13]
## Picking random jokes often picked the joke already picked on the next random pick. 
___
### To make it really random, follow these...
- Create a collection of picked random jokes
- If a random joke query is made, select a random joke
- Check if the selected joke is already selected recently or present in the collection
- If present, pick another random joke and check again if present, if not return the picked joke. 
- Reset the collection if all the random jokes already picked.

**Comment by Anzo52** [comment]
I could give it a try if you'd like. I was looking at the JokeService class and was thinking that generateRandomJoke() should be in a separate class from the record stuff. I'm trying to learn more architecture-related stuff so any thoughts on that would be appreciated.

**Comment by pitzzahh** [comment]
Hi, sorry for late response, just got back from school. Thanks for giving your response about my project. I am currently new to building APIs with Spring and it's architecture, I am open for help, you can do the thing that you suggest which is to separate the method in the record JokeService, well I just used record because of it's features such as no need for accessors and auto-wiring will be easy. You can fork the main branch and create a PR on your changes. Thanks

**Comment by pitzzahh** [comment]
fix unused Category parameter in method. Found on commit: 8618456b29f1613e67da4c70b74cf61fa477b72d

**Comment by Anzo52** [comment]
Another thought: how about an sqlite database for the jokes? Then just have a view table that calls ORDER BY RANDOM(). I think it would be slightly more elegant than relying on Math.random() (which, if I'm not mistaken, isn't technically random, but maybe that's Python's version). Also, in my (quite inexperienced) opinion, SQL is a lot easier to work with than JSON.
Am I over-complicating the solution or do you think it would work?

**Comment by pitzzahh** [comment]
I think it will work. The JSON jokes are only temporary so that I can test. If implemented this every-time the endpoint was called there will be a call to the database which might cause performance issues. In my solution I want to load all the jokes initially from the database(any db) then load it to a collection, then when the endpoint was called, then it won't need to a query on the db because we can now use the collection that contains all the jokes(I found this faster than making a query every-time)

**Comment by pitzzahh** [comment]
I actually postponed this because I have an urgent project. But, this is still wip

**Comment by Anzo52** [comment]
Yeah, that makes much more sense using H2, actually. I'll play around with it a bit more.

**Comment by pitzzahh** [comment]
Yes, H2 is good when doing testing stuff
